redis/release_lock.lua

```diff
@@ -4,9 +4,24 @@ local stored_jid = redis.pcall('get', unique_key)
 
 if stored_jid then
   if stored_jid == job_id or stored_jid == '2' then
-    redis.pcall('del', unique_key)
-    redis.pcall('hdel', 'uniquejobs', job_id)
-    return 1
+    if redis.pcall('del', unique_key) then
+      if redis.pcall('hdel', 'uniquejobs', job_id) then
+        -- All should be good
+        return 1
+      else
+        -- hdel has failed
+        return 2
+      end
+    else
+      -- del has failed, lets try hdel anyway.
+      if redis.pcall('hdel', 'uniquejobs', job_id) then
+        -- All should be good anyway ?
+        return 3
```

Comment:
this case is currently considered a succes, but maybe it should not... 
something we could/should do here?